core/CronArchive/QueueConsumer.php

```diff
@@ -201,6 +201,11 @@ public function getNextArchivesToProcess()
                 break;
             }
 
+            if (!$this->isSegmentAvaible($invalidatedArchive)) {
+                $this->logger->debug("Segment is not valid anymore.");
+                break;
```

Comment:
Actually I don't think that a break is correct here. That would actually stop the whole invalidation queue. It should more likely be a `continue`. In addition it could make sense to remove the invalidation, so it doesn't happen in each run. Similar to the block below, where it checks if the plugin is still active.

Reply:
The removal of the invalidation is still missing. The other changes look fine.